jupyter_publishing_service/app.py

```diff
@@ -88,18 +93,25 @@ def _validate_ip(self, proposal):
             value = ""
         return value
 
+    authenticator: AuthorizerABC = Instance(
+        klass="jupyter_publishing_service.authenticator.abc.AuthenticatorABC", allow_none=True
+    )
+
+    storage_manager: StorageManagerABC = Instance(
+        klass="jupyter_publishing_service.storage.abc.StorageManagerABC", allow_none=True
+    )
+
     def init_configurables(self):
-        self.file_manager = self.file_manager_class(parent=self, log=self.log)
-        self.collaborator_store = self.collaborator_store_class(parent=self, log=self.log)
         self.authenticator = self.authenticator_class(parent=self, log=self.log)
-        self.authorizer = self.authorizer_class(parent=self, log=self.log)
         set_authenticator_class(self.authenticator)
-        set_authorizer_class(self.authorizer)
+        self.storage_manager = self.storage_manager_class(parent=self, log=self.log)
+        self.storage_manager.initialize()
 
     def init_webapp(self):
         self.app = FastAPI(
-            title="Jupyter Publishing Server",
-            description="Jupyter File Publishing Server implementation powered by FastAPI.",
+            title=TITLE,
+            description=SUMMARY,
+            lifespan=lifespan,
```

Comment:
Suspect this may fail as I have seen `router.app.storage_manager` to only resolve after `self.app.include_router(router)` and `router.app = self` is invoked. Curious to learn if there is a gap in the understanding.

Reply:
Hm, I don't see this fail. 

I think it's because `lifespan` doesn't execute until after the app is actually served, which in this case, happens after the next two lines.